website/addons/base/__init__.py

```diff
@@ -369,13 +370,22 @@ def grant_oauth_access(self, node, external_account, metadata=None):
 
         self.save()
 
-    def revoke_oauth_access(self, external_account):
+    @must_be_logged_in
+    def revoke_oauth_access(self, external_account, auth):
         """Revoke all access to an ``ExternalAccount``.
 
         TODO: This should accept node and metadata params in the future, to
             allow fine-grained revocation of grants. That's not yet been needed,
             so it's not yet been implemented.
         """
+        for node in self.get_nodes_with_oauth_grants(external_account):
+            try:
+                addon_settings = node.get_addon(external_account.provider)
```

Comment:
It would be raised on `.get_addon` if the oauth grant exists but the `[Provider]NodeSettings` object doesn't, which is a possible but edge configuration for a node.